### ISSUE TITLE
Latest version failing on conflict contol

### ISSUE THREAD

**Issue by n1koo** [GoogleCloudPlatform/k8s-config-connector#96]
Getting `"invalid management conflict policy '', can be one of {none, resource, project}"`

Full error:
```
{"level":"error","ts":1582623253.4698217,"logger":"kubebuilder.controller","msg":"Reconciler error","controller":"redisinstance-controller","request":"helloms-test-sit/helloms-niko","error":"invalid management conflict policy '', can be one of {none, resource, project}","stacktrace":"cnrm.googlesource.com/cnrm/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/src/cnrm.googlesource.com/cnrm/vendor/github.com/go-logr/zapr/zapr.go:128\ncnrm.googlesource.com/cnrm/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/src/cnrm.googlesource.com/cnrm/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:217\ncnrm.googlesource.com/cnrm/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1\n\t/go/src/cnrm.googlesource.com/cnrm/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158\ncnrm.googlesource.com/cnrm/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/go/src/cnrm.googlesource.com/cnrm/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\ncnrm.googlesource.com/cnrm/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/src/cnrm.googlesource.com/cnrm/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134\ncnrm.googlesource.com/cnrm/vendor/k8s.io/apimachinery/pkg/util/wait.Until\n\t/go/src/cnrm.googlesource.com/cnrm/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88"}
```

on latest version `controller:b06a1e6` . With previous version `aa4075b` things are still fine.

Was going through the CRD changes in latest release package, but cant spot this to be a CRD change, most likely a controller config is needed. 

Any pointers to config i'm missing (eg. where to set conflict policy and docs on it) ?

**Comment by spew** [comment]
Hi @n1koo, you can fix this by applying the `cnrm.cloud.google.com/management-conflict-prevention-policy` annnotation to your resource. It can be done as follows:

```
metadata:
  annotations:
    cnrm.cloud.google.com/management-conflict-prevention-policy: none```

**Comment by n1koo** [comment]
Thanks for the quick reply! Tried it out in sandbox env and works, but i'll wait until the docs are out before rolling out elsewhere. 

I also assume it makes sense for this to become a default set by controller..?

**Comment by spew** [comment]
Hi @n1koo, it is set by default by an admission handler, unfortunately this of course doesn't apply to resources that area already stored in etcd. Your suggested change is being made now and will be in this week's release.

**Comment by spew** [comment]
Hi @n1koo this issue should be resolved with the latest release from this week.

**Comment by kibbles-n-bytes** [comment]
Hey @n1koo , I'm going to close out this issue as the fix is in from our 1.3.0 release onwards. If you are still having problems, feel free to open another issue.
